Mathlib/Geometry/Manifold/InteriorBoundary.lean

```diff
@@ -123,36 +125,42 @@ instance : BoundarylessManifold I M where
     let r := ((chartAt H x).isOpen_extend_target I).interior_eq
     have : extChartAt I x = (chartAt H x).extend I := rfl
     rw [← this] at r
-    rw [ModelWithCorners.isInteriorPoint_iff, r]
+    rw [isInteriorPoint_iff, r]
     exact PartialEquiv.map_source _ (mem_extChartAt_source _ _)
 
 end Boundaryless
 
 section BoundarylessManifold
-variable [BoundarylessManifold I M]
 
 /-- The empty manifold is boundaryless. -/
 instance BoundarylessManifold.of_empty [IsEmpty M] : BoundarylessManifold I M where
   isInteriorPoint' x := (IsEmpty.false x).elim
 
-lemma _root_.BoundarylessManifold.isInteriorPoint {x : M} :
+lemma _root_.BoundarylessManifold.isInteriorPoint {x : M} [BoundarylessManifold I M] :
     IsInteriorPoint I x := BoundarylessManifold.isInteriorPoint' x
 
 /-- If `I` is boundaryless, `M` has full interior. -/
-lemma interior_eq_univ : I.interior M = univ :=
+lemma interior_eq_univ [BoundarylessManifold I M] : I.interior M = univ :=
   eq_univ_of_forall fun _ => BoundarylessManifold.isInteriorPoint I
 
 /-- Boundaryless manifolds have empty boundary. -/
-lemma Boundaryless.boundary_eq_empty : I.boundary M = ∅ := by
+lemma Boundaryless.boundary_eq_empty [BoundarylessManifold I M] : I.boundary M = ∅ := by
   rw [I.boundary_eq_complement_interior, I.interior_eq_univ, compl_empty_iff]
 
 instance [BoundarylessManifold I M] : IsEmpty (I.boundary M) :=
   isEmpty_coe_sort.mpr (Boundaryless.boundary_eq_empty I)
 
+/-- Manifolds with empty boundary are boundaryless. -/
+lemma Boundaryless.of_boundary_eq_empty (h : I.boundary M = ∅) : BoundarylessManifold I M where
+  isInteriorPoint' x := by
+    show x ∈ I.interior M
+    rw [boundary_eq_complement_interior, compl_empty_iff] at h
```

Comment:
I think that `boundary_eq_complement_interior` should be called `ModelWithCorners.compl_interior`, with swapped LHS and RHS.

Reply:
Done